### DISCUSSION TITLE
cURL fails authentication for sending email

### DISCUSSION THREAD

**Discussion by OpenShooter** [curl/curl#12941] [Q&A]
I'm updating an old C++ pgm that uses `libcurl`. When testing the ability to send email, I discovered that the login to the sending mail server is failing. I think this must be related to the changes ISPs made in authentication (my ISP is AT&T, which uses Yahoo). Apparently a user name + password is not sufficient for login, even though two-factor authentication is not enabled for my account.

To use my email client (Eudora) to log into my email account, I use a "secure key" previously generated by the ISP as the email client's password. If I use the old 10-character password which was valid in 2013, cURL returns error `CURLE_LOGIN_DENIED` (67), "user, password or similar was not accepted and we failed to login". If I use the new 16-character secure key, I get error `CURLE_WEIRD_SERVER_REPLY` (8).
My send-an-email code follows the example for `curl_easy_perform`.

What changes do I need to make to my C++ source code so I can send an email? The usage of my pgm is such that only a few emails will be allowed as senders or destinations, hence I could require those email accounts to be restricted in some way.

Windows 11, Visual Studio 2022, cURL 8.6.0

**Comment by dfandrich** [comment]
You'll have to find out exactly how that key is supposed to be used in order to
get libcurl to use it that way. Chances are, its supposed to be used with
CURLOPT_LOGIN_OPTIONS or CURLOPT_XOAUTH2_BEARER or something like that and with
the traditional normal user/password mechanism.
